src/admin/components/forms/field-types/RichText/elements/upload/Button/index.tsx

```diff
@@ -131,6 +154,35 @@ const UploadButton: React.FC<{ path: string }> = ({ path }) => {
                   {' '}
                   {modalCollection.labels.singular}
                 </h1>
+                <Pill onClick={() => {
+                  toggleModal(uploadModalSlug);
+                  setRenderModal(true);
+                }}>
+                  Create New
+                </Pill>
+                {renderModal && (
+                  <Modal className={baseModalClass} slug={uploadModalSlug}>
+                    <MinimalTemplate width="wide">
+                      <header className={`${baseModalClass}__header`}>
+                      <h1>Upload</h1>
+                      <Button
+                        icon="x"
+                        round
+                        buttonStyle="icon-label"
+                        iconStyle="with-border"
+                        onClick={() => {
+                          toggleModal(uploadModalSlug);
+                          setRenderModal(false);
+                        }}
+                      />
+                      </header>
+                      <Form onSubmit={handleAddUpload} disabled={loading}>
+                        <Upload data={data} collection={modalCollection} />
+                        <Submit>Add new Image</Submit>
```

Comment:
Hey @Sccttt — this is a great PR. Only thing I am seeing before pulling this down and reviewing manually is that we are not rendering the _**fields**_ that are associated with the upload collection here, and we need to, as some fields on an upload-enabled collection may be required / similar.

Take a look at how we do it in the `upload` field type here:
https://github.com/payloadcms/payload/blob/master/src/admin/components/forms/field-types/Upload/Add/index.tsx#L84

Can you make sure that you render the corresponding fields for the collection here as well? Then I'll pull this down and do some testing!